src/main/java/org/folio/inventory/resources/IsbnUtilsApi.java

```diff
@@ -0,0 +1,157 @@
+        ? isbnCode
+        : IsbnUtil.convertTo10DigitNumber(isbnCode);
+
+    convert(routingContext, toIsbn10Converter);
+  }
+
+  private void convert(RoutingContext routingContext, UnaryOperator<String> converter) {
+
+    String isbnCode = EMPTY;
+
+    try {
+
+      isbnCode = getIsbnParamValue(routingContext);
+      boolean isHyphens = getHyphensParamValue(routingContext);
+
+      isbnCode = Optional.ofNullable(converter.apply(isbnCode))
+        .orElseThrow(() -> new ISBNException(INVALID_ISBN_MESSAGE));
+
+      isbnCode = insertHyphens(isbnCode, isHyphens);
+
+      JsonObject result = new JsonObject();
+      result.put(ISBN_PARAM, isbnCode);
+      JsonResponse.success(routingContext.response(), result);
+
+    } catch (ISBNException e) {
+      log.error(e);
+      ValidationError error = new ValidationError(e.getMessage(), ISBN_PARAM, isbnCode);
+      JsonResponse.badRequest(routingContext.response(), error);
+    } catch (IllegalArgumentException e) {
+      log.error(e);
+      ClientErrorResponse.badRequest(routingContext.response(), e.getMessage());
+    }
+  }
+
+  private String getIsbnParamValue(RoutingContext routingContext) {
+    String isbnCode;
+    if (isEmpty(routingContext.queryParam(ISBN_PARAM))) {
+      throw new IllegalArgumentException(CONVERTER_MISSING_REQUIRED_PARAM_MSG);
+    }
+
+    isbnCode = routingContext.queryParam(ISBN_PARAM).get(0);
+    return isbnCode;
+  }
+
+  private boolean getHyphensParamValue(RoutingContext routingContext) {
+    if (isNotEmpty(routingContext.queryParam(HYPHENS_PARAM))) {
+      String hyphensValue = routingContext.queryParam(HYPHENS_PARAM).get(0);
+
+      if ("true".equalsIgnoreCase(hyphensValue) || "false".equalsIgnoreCase(hyphensValue)) {
+        return Boolean.valueOf(hyphensValue);
+      }
+      throw new IllegalArgumentException(INVALID_HYPHENS_VALUE_MSG);
+    }
+    return false;
+  }
+
+  private String insertHyphens(String isbnCode, boolean isHyphens) {
+    if (isHyphens) {
+      isbnCode = IsbnUtil.insertHyphens(isbnCode);
+    }
+    return isbnCode;
+  }
+
+  private void validate(RoutingContext routingContext) {
+
+    JsonObject result = new JsonObject();
+    boolean isValid;
+    String paramName = EMPTY;
+    String isbnCode = EMPTY;
+
+    MultiMap params = routingContext.queryParams();
+
+    if (params.entries().size() == 1) {
+      Map.Entry<String, String> param = params.entries().get(0);
+      paramName = param.getKey();
+      isbnCode = param.getValue();
+    }
+
+    switch (paramName) {
+      case "isbn10":
+        isValid = IsbnUtil.isValid10DigitNumber(isbnCode);
+        break;
+      case "isbn13":
+        isValid = IsbnUtil.isValid13DigitNumber(isbnCode);
+        break;
+      case ISBN_PARAM:
+        isValid = IsbnUtil.isValid10DigitNumber(isbnCode) || IsbnUtil.isValid13DigitNumber(isbnCode);
+        break;
+      default: ClientErrorResponse.badRequest(routingContext.response(),
```

Comment:
Instead of plain text, please return error json

Reply:
Done